docs/coding.md

```diff
@@ -0,0 +1,102 @@
+# Coding Conventions
+
+Please follow coding conventions and guidelines described in the following documents:
+
+* [Go proverbs](https://go-proverbs.github.io/) - highly recommended read
+* [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments)
+* [Effective Go](https://golang.org/doc/effective_go.html)
+* [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+
+Here's a list of some more specific conventions that are often followed in
+the code and will be pointed out in the review process:
+
+## General
+
+* Keep variable names short for variables that are local to the function.
+* Do not export a function or variable name outside the package until you
+  have an external consumer for it.
```

Comment:
An example of how hard breaks in the md file breaks readability
![image](https://github.com/ceph/ceph-csi-operator/assets/14993359/b15719d4-6120-47d8-b8b4-5ebab024c943)

Can we please remove the hard break from this line?